modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/sync/data/EpisodeSync.kt

```diff
@@ -0,0 +1,151 @@
+        return withContext(Dispatchers.Default) {
+            episodes.map { localEpisode ->
+                record {
+                    episode = syncUserEpisode {
+                        uuid = localEpisode.uuid
+                        podcastUuid = localEpisode.podcastUuid
+                        localEpisode.playingStatusModified?.let { modifiedAt ->
+                            playingStatus = int32Value { value = localEpisode.playingStatus.toInt() }
+                            playingStatusModified = int64Value { value = modifiedAt }
+                        }
+                        localEpisode.starredModified?.let { modifiedAt ->
+                            starred = boolValue { value = localEpisode.isStarred }
+                            starredModified = int64Value { value = modifiedAt }
+                        }
+                        localEpisode.playedUpToModified?.let { modifiedAt ->
+                            playedUpTo = int64Value { value = localEpisode.playedUpTo.toLong() }
+                            playedUpToModified = int64Value { value = modifiedAt }
+                        }
+                        localEpisode.durationModified?.takeIf { localEpisode.duration != 0.0 }?.let { modifiedAt ->
+                            duration = int64Value { value = localEpisode.duration.toLong() }
+                            durationModified = int64Value { value = modifiedAt }
+                        }
+                        localEpisode.archivedModified?.let { modifiedAt ->
+                            isDeleted = boolValue { value = localEpisode.isArchived }
+                            isDeletedModified = int64Value { value = modifiedAt }
+                        }
+                        localEpisode.deselectedChaptersModified?.let { modifiedAt ->
+                            deselectedChapters = ChapterIndices.toString(localEpisode.deselectedChapters)
+                            deselectedChaptersModified = int64Value { value = modifiedAt.time }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    suspend fun processIncrementalResponse(serverEpisodes: List<SyncUserEpisode>) {
+        val serverEpisodesMap = serverEpisodes.associateBy(SyncUserEpisode::getUuid)
+        val localEpisodes = episodeManager.findByUuids(serverEpisodesMap.keys)
+
+        val episodesToArchive = mutableListOf<PodcastEpisode>()
+        val episodeToFinish = mutableListOf<PodcastEpisode>()
+        localEpisodes.forEach { localEpisode ->
+            val serverEpisode = serverEpisodesMap[localEpisode.uuid] ?: return@forEach
+            localEpisode.applyServerEpisode(
+                serverEpisode,
+                onShouldBeArchived = { episodesToArchive += localEpisode },
+                onShouldBeFinished = { episodeToFinish += localEpisode },
+            )
+        }
+        episodeManager.archiveAllInList(episodesToArchive, playbackManager)
```

Comment:
I don't believe this will set the `archived_modified` to NULL so it keeps trying to sync the episodes. I noticed it in my user admin area logs. I think this is also happening with the playing status.